homeassistant/components/airnow/__init__.py

```diff
@@ -44,10 +44,12 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up AirNow from a config entry."""
-    api_key = entry.data[CONF_API_KEY]
-    latitude = entry.data[CONF_LATITUDE]
-    longitude = entry.data[CONF_LONGITUDE]
-    distance = entry.data[CONF_RADIUS]
+    api_key = entry.data.get(CONF_API_KEY)
+    latitude = entry.data.get(CONF_LATITUDE)
+    longitude = entry.data.get(CONF_LONGITUDE)
+
+    # Station Radius can come from configuration or options
+    distance = entry.options.get(CONF_RADIUS, entry.data.get(CONF_RADIUS))
```

Comment:
We can move it to options instead of having it in both places

Reply:
Also that requires it to be set during config flow for new entries.

Reply:
Is there documentation or best practice example on how this is supposed to be done?  I don't fully understand how `entry.options` and `entry.data` interact, and this structure was copied from an existing integration.

Reply:
`data` and `options` are only two places (dicts) where you can store information while obviously the purpose is that `options` can change while `data` doesn't.
So what I propose is that in __init__ you remove the key from `data` and set it in `options` instead so when the user goes to edit options there is already a default there (since it was in from before).

Reply:
Sorry, I'm still not completely following where to copy the key from `data` to `options`.  AIUI, when the integration is first set up, the `options` dict is empty when `async_setup_entry` first gets called, so to do it there I would still have to test whether `options` already had the key, which is effectively what my existing code is doing anyway.

I do see that I can pass an `options` parameter to `ConfigFlow.async_create_entry` which seems to then populate the `options` dict for `async_setup_entry` - is that a workable way to get to the same goal?

```python
return self.async_create_entry(
    title=(
        f"AirNow Sensor at {user_input[CONF_LATITUDE]},"
        f" {user_input[CONF_LONGITUDE]}"
    ),
    data=user_input,
    options={ CONF_RADIUS: user_input[CONF_RADIUS] },
)
```

Reply:
> I do see that I can pass an options parameter to ConfigFlow.async_create_entry which seems to then populate the options dict for async_setup_entry - is that a workable way to get to the same goal?

Yes. But for existing users you need to use `hass.config_entries.async_update_entry()` to modify `data` and `options` before continuing. That needs to happen in `async_setup_entry()` in `__init__`

Reply:
OK, makes sense.  I think I have it fixed up now.